### PR TITLE
Add keyboard shortcuts for the debugger

### PR DESCRIPTION
**Pull request by jasonLaster**
This is a follow up PR from #540.

I'll file an `m-c` issue soon to address the change

### CODE REVIEW

File: public/js/components/RightSidebar.js [hunk 1]
```diff
@@ -41,33 +41,65 @@ const RightSidebar = React.createClass({
...
+
+  componentWillUnmount() {
+    const { keyShortcuts } = this.props;
+    keyShortcuts.off("F8", this.resume);
+    keyShortcuts.off("F10", this.stepOver);
+    keyShortcuts.off("F11", this.stepIn);
+    keyShortcuts.off("F12", this.stepOut);
+  },
 
+  renderStepButtons() {
```
Comment (jasonLaster) [inline public/js/components/RightSidebar.js]: it's hard to see here, but adding the individual step functions cleaned up this function quite a bit.

**Comment by jbhoosreddy** [comment]
@jasonLaster, thanks for taking care of this.

**Comment by jbhoosreddy** [comment]
Also, in regards to the change in m-c, I find it a bit strange that there is a bug in m-c library. Because the existing keyboard shortcuts work in the current debugger. I'm curious to see if you run into a similar problem.

**Comment by jlongster** [comment]
> Also, in regards to the change in m-c, I find it a bit strange that there is a bug in m-c library. Because the existing keyboard shortcuts work in the current debugger. I'm curious to see if you run into a similar problem.

That's exactly what I'm curious about. We should investigate this before making changes to m-c and understand why we need to make those changes.

**Comment by jasonLaster** [comment]
It looks like the toolbox is using a different mechanism for registering the key shortcuts:

screen shot 2016-08-17 at 5 34 56 pm


https://dxr.mozilla.org/mozilla-central/source/devtools/client/debugger/views/toolbar-view.js#55-79

https://dxr.mozilla.org/mozilla-central/source/devtools/client/debugger/debugger.xul#183-216

**Comment by jlongster** [comment]
Makes sense, thanks!

**Pull request merged**